### PR TITLE
VACMS-4281: upgrade migrate plus + drush and add migration tests

### PR DESCRIPTION
**Pull request by acrollet**
## Description

See #4281 

## QA steps

As user _uid_ with _user_role_
1. Do this
1. Then that
1. Then validate Acceptance Criteria from issue
- [ ] This
- [ ] That
- [ ] The other thing

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Core Application Team`
- [ ] `Product Support Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: tests/phpunit/Migration/VaFormMigrationTest.php [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  /**
+   * Test the VA Form Migration.
+   *
+   * This test first imports a new form and verifies that a new node is created.
+   * It then re-runs the same migration with updated data and verifies that the
+   * updated data has been saved to the node.
+   *
+   * @dataProvider vaFormDataProvider
+   */
+  public function testVaFormMigration(
```
Comment (acrollet) [inline tests/phpunit/Migration/VaFormMigrationTest.php:67]: @swirtSJW I'd appreciate it if you could look over this test when you're able and let me know if there's anything you'd like to add. In particular, if you want to validate any of the data processing/sanitization that's going on. Thanks!

**Comment by acrollet** [comment]
This PR is blocked by #3868 as migrate_tools 5.x does not support drush 8.

File: tests/phpunit/Support/Entity/Storage.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
+<?php
```
Comment (ndouglas) [inline tests/phpunit/Support/Entity/Storage.php:1]: This definitely makes the tests a lot more readable and much less tedious to write.

File: tests/phpunit/Support/Mock/HttpClient.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<?php
```
Comment (ndouglas) [inline tests/phpunit/Support/Mock/HttpClient.php:1]: Great class for easing these one-off cases.

File: tests/phpunit/Support/Migration/Migrator.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
+<?php
```
Comment (ndouglas) [inline tests/phpunit/Support/Migration/Migrator.php:1]: Nice.  Eliminates a lot of boilerplate.

**Review by ndouglas — COMMENTED** [review]
Great test improvements!

**Comment by acrollet** [comment]
Replaced by #4590

**Pull request closed without merging**